front/app/utils/currency/FormattedCurrency.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import useAppConfiguration from 'hooks/useAppConfiguration';
+import { isNilOrError } from 'utils/helperUtils';
```

Comment:
I like that you did currency and budget separately and didn't try to make it a single component 👍

Reply:
Oh, FormattedBudget (the other component) also adds the currency. Was that not clear? I might have to change/clarify it then.

Reply:
Oh, ok! I see it now. But then, why do we need both?

Reply:
I will use FormattedCurrency in a different branch. There we have the form where admins specify e.g. minimum budget and then we like to show their currency there so they know what they are specifying. It's in a way a validation that they've correctly installed their currency to be 'tokens' or 'credits' instead of only finding out afterwards when people can already submit budgets. Now, the problem is, with credits and tokens we don't want to show CRE or TOK as that would be unclear. So I can't just use the currency defined in the core settings. Hence I made the FormattedCurrency (name could still change) component to display 'tokens' or 'credits' when we have CRE or TOK as currency (more custom ones can be added).

We also need FormattedBudget, for when we want to format the currency units (for which we use FormattedNumber inside FormattedBudget). E.g. 10000 will become 10.000 €).

Reply:
Ok, I see! Perfect then, it looks good 👍